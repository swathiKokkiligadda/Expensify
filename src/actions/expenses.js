import { v4 as uuidv4 } from "uuid";
//Add expense action generator
export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0,
} = {}) => ({
  type: "ADD_EXPENSE",
  expense: {
    id: uuidv4(),
    description,
    note,
    amount,
    createdAt,
  },
});

//remove expense action generator
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id: id,
});
//edit expense action generator
export const editExpense = (id, updates) => ({
  type: "EDIT_EXPENSE",
  id,
  updates,
});

