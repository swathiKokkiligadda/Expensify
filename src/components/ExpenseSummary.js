import React from 'react';
import selectedExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import {connect} from 'react-redux';
import numeral from 'numeral';

export const ExpenseSummary = ({expenseCount, expensesTotal}) => {
  const expenseWord = expenseCount ===1?'expense':'expenses';
  const formattedExpenseTotal = numeral(expensesTotal/100).format('$0,0.00');
  return (
    <div>
    <h1>Viewing {expenseCount} {expenseWord} totaling {formattedExpenseTotal}</h1>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectedExpenses(state.expenses, state.filters);
  return{
    expenseCount:visibleExpenses.length,
    expensesTotal:expensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpenseSummary);
