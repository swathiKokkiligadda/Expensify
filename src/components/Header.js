import { NavLink } from "react-router-dom";
import React from "react";
const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>
      Dashboard
    </NavLink>
    <br />
    <NavLink to="/create" activeClassName="is-active">
      Create Expense
    </NavLink>
    <br />
    <NavLink to="/help" activeClassName="is-active">
      Help
    </NavLink>
    <br />
  </header>
);
export default Header;
