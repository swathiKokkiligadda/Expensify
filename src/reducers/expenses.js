//expense reducer
const expensesReducerDefault = [];

export default (state = expensesReducerDefault, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      return state.filter((item) => {
        return item.id !== action.payload;
      });
    case "EDIT_EXPENSE":
      return state.map((expense) => {
        if (expense.id === action.id) {
          return { ...expense, ...action.updates };
        } else {
          return expense;
        }
      });

    default:
      return state;
  }
};
