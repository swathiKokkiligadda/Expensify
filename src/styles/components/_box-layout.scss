.box-layout {
  align-items: center;
  background: url("/images/bg.jpg");
  height: 100vh;
  width: 100vw;
  display: flex;
  justify-content: center;
  background-size: cover;
}
.box-layout__box {
  background: fade-out($color: White, $amount: 0.2);
  border-radius: 3px;
  text-align: center;
  width: 25rem;
  padding: $l-size $m-size;
}
.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
