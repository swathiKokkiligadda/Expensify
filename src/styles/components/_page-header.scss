.page-header {
  background-color: $off-white;
  padding: $l-size 0;
  margin-bottom: $l-size;
}
.page-header__title {
  margin: 0;
  font-weight: 300;
}
.page-header__actions {
  margin: $m-size;
}
span {
  font-weight: bold;
}
