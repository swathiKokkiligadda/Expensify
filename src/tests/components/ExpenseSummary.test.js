import React from 'react';
import {ExpenseSummary} from '../../components/ExpenseSummary';
import {shallow} from 'enzyme'
import expenses from '../fixtures/expenses';

test('should correctly render expensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal ={235}/>);
  expect(wrapper).toMatchSnapshot();
})

test('should correctly render expensesSummary with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={23} expensesTotal ={233435}/>);
  expect(wrapper).toMatchSnapshot();
})