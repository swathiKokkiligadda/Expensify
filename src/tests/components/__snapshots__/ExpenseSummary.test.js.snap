// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render expensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.35
  </h1>
</div>
`;

exports[`should correctly render expensesSummary with 1 expense 2`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totaling 
    $2,334.35
  </h1>
</div>
`;
